app/services/omniauth_user_resolver.rb

```diff
@@ -106,11 +108,27 @@ def omniauth_provider
   end
 
   def first_name
-    @first_name ||= auth.info.first_name
+    @first_name ||= names[:first]
   end
 
   def last_name
-    @last_name ||= auth.info.last_name
+    @last_name ||= names[:last]
+  end
+
+  def names
+    @names ||= extract_first_and_last_name
+  end
+
+  def extract_first_and_last_name
```

Comment:
I'm not 100% convinced of the robustness of this.
what if the first name is something like
Jean Luc